controllers/perconaservermysql_controller.go

```diff
@@ -1563,3 +1569,160 @@ func (r *PerconaServerMySQLReconciler) restartGroupReplication(ctx context.Conte
+			return errors.Wrap(err, "create CA certificate")
+		}
+
+		if err := r.waitForCerts(ctx, cr.Namespace, cr.Name+"-ca-cert"); err != nil {
+			return err
+		}
+
+		issuerConf = cm.IssuerConfig{
+			CA: &cm.CAIssuer{SecretName: caCert.Spec.SecretName},
+		}
+
+		if err := r.createIssuer(ctx, cr, issuerName, issuerConf); err != nil {
+			return err
+		}
+	}
+	hosts := []string{
+		fmt.Sprintf("*.%s-mysql", cr.Name),
+		fmt.Sprintf("*.%s-mysql.%s", cr.Name, cr.Namespace),
+		fmt.Sprintf("*.%s-mysql.%s.svc.cluster.local", cr.Name, cr.Namespace),
+		fmt.Sprintf("*.%s-orchestrator", cr.Name),
+		fmt.Sprintf("*.%s-orchestrator.%s", cr.Name, cr.Namespace),
+		fmt.Sprintf("*.%s-orchestrator.%s.svc.cluster.local", cr.Name, cr.Namespace),
+		fmt.Sprintf("*.%s-router", cr.Name),
+	}
+	kubeCert := &cm.Certificate{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      cr.Name + "-ssl",
+			Namespace: cr.Namespace,
+		},
+		Spec: cm.CertificateSpec{
+			SecretName: cr.Spec.SSLSecretName,
+			DNSNames:   hosts,
+			IsCA:       false,
+			IssuerRef: cmmeta.ObjectReference{
+				Name:  issuerName,
+				Kind:  issuerKind,
+				Group: issuerGroup,
+			},
+		},
+	}
+	err := ctrl.SetControllerReference(cr, kubeCert, r.Scheme)
+	if err != nil {
+		return errors.Wrap(err, "set controller refference")
+	}
+	if cr.Spec.TLS != nil {
+		kubeCert.Spec.DNSNames = append(kubeCert.Spec.DNSNames, cr.Spec.TLS.SANs...)
+	}
+
+	err = r.Create(ctx, kubeCert)
+	if err != nil && !k8serrors.IsAlreadyExists(err) {
+		return errors.Wrap(err, "create certificate")
+	}
+
+	return r.waitForCerts(ctx, cr.Namespace, cr.Spec.SSLSecretName)
+}
+
+func (r *PerconaServerMySQLReconciler) waitForCerts(ctx context.Context, namespace string, secretsList ...string) error {
+	ticker := time.NewTicker(3 * time.Second)
+	timeoutTimer := time.NewTimer(30 * time.Second)
+	defer timeoutTimer.Stop()
+	defer ticker.Stop()
+	for {
+		select {
+		case <-timeoutTimer.C:
+			return errors.Errorf("timeout: can't get tls certificates from certmanager, %s", secretsList)
+		case <-ticker.C:
+			sucessCount := 0
+			for _, secretName := range secretsList {
+				secret := &corev1.Secret{}
+				err := r.Get(ctx, types.NamespacedName{
+					Name:      secretName,
+					Namespace: namespace,
+				}, secret)
+				if err != nil && !k8serrors.IsNotFound(err) {
+					return err
+				} else if err == nil {
+					sucessCount++
+				}
+			}
+			if sucessCount == len(secretsList) {
+				return nil
+			}
+		}
+	}
+}
+
+func (r *PerconaServerMySQLReconciler) createIssuer(ctx context.Context, cr *apiv1alpha1.PerconaServerMySQL, issuerName string, IssuerConf cm.IssuerConfig,
+) error {
+
+	isr := &cm.Issuer{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      issuerName,
+			Namespace: cr.Namespace,
+		},
+		Spec: cm.IssuerSpec{
+			IssuerConfig: IssuerConf,
+		},
+	}
+	err := ctrl.SetControllerReference(cr, isr, r.Scheme)
+	if err != nil {
+		return errors.Wrap(err, "set controller refference")
```

Comment:
```suggestion
		return errors.Wrap(err, "set controller reference")
```
There are also the same typos in this file

Reply:
https://github.com/percona/percona-server-mysql-operator/pull/125/commits/5b6d2322a1332dcda990dd7b58e91543632a91fc
https://github.com/percona/percona-server-mysql-operator/pull/125/commits/6344dd58dc22a7bfad05ae84ed3f70ef96e5f6ac